src/Gui/MainWindow.cpp

```diff
@@ -2222,6 +2231,121 @@ void MainWindow::showStatus(int type, const QString& message)
+    showHints({ hint });
+}
+
+void MainWindow::showHints(const std::list<InputHint>& hints)
+{
+    if (hints.empty()) {
+        hideHint();
+        return;
+    }
+
+    static const auto maybeIconPath = [](const InputHint::UserInput key) -> std::optional<const char*> {
+        switch (key) {
+            case InputHint::UserInput::MouseLeft:
+                return ":/icons/keys/mouse-left.svg";
+            case InputHint::UserInput::MouseRight:
+                return ":/icons/keys/mouse-right.svg";
+            case InputHint::UserInput::MouseMove:
+                return ":/icons/keys/mouse-move.svg";
+            default:
+                return std::nullopt;
+        }
+    };
+
+    static const auto generateKeyIcon = [](const InputHint::UserInput key,
+                                           const QColor color) -> QPixmap {
+        constexpr int margin = 3;
+        constexpr int padding = 4;
+        constexpr int radius = 2;
+        constexpr int iconTotalHeight = 24;
+        constexpr int iconSymbolHeight = iconTotalHeight - 2*margin;
+
+        const QFont font(QStringLiteral("sans"), 10, QFont::Bold);
+        const QFontMetrics fm(font);
+        const QString text = QString::fromUtf8(stringifyUserInput(key));
+        const QRect textBoundingRect = fm.tightBoundingRect(text);
+
+        const auto symbolWidth = std::max<unsigned>(textBoundingRect.width() + padding * 2, iconSymbolHeight);
+
+        QImage image(symbolWidth + margin * 2, iconTotalHeight, QImage::Format_ARGB32_Premultiplied);
+        image.fill(0x00000000);
+
+        const QRect keyRect(margin, margin, symbolWidth, 18);
+
+        QPainter painter(&image);
+        painter.setRenderHint(QPainter::Antialiasing);
+        painter.setPen(QPen(color, 2));
+        painter.setFont(font);
+        painter.drawRoundedRect(keyRect, radius, radius);
+        painter.drawText(
+            // adjust the rectangle so it is visually centered, for characters that are below baseline
+            keyRect.translated(0, -(textBoundingRect.y() + textBoundingRect.height()) / 2),
+            Qt::AlignHCenter,
+            text);
+
+        return QPixmap::fromImage(image);
+    };
+
+    const auto getKeyImage = [this](InputHint::UserInput key) {
+        const auto& factory = BitmapFactory();
+
+        QPixmap image = [&] {
+            QColor color = d->hintLabel->palette().text().color();
+
+            if (auto iconPath = maybeIconPath(key)) {
+                return factory.pixmapFromSvg(*iconPath,
+                                             QSize(24, 24),
+                                             {{0xFFFFFF, color.rgb() & RGB_MASK}});
+            }
+
+            return generateKeyIcon(key, color);
+        } ();
+
+
+        QBuffer buffer;
+        image.save(&buffer, "png");
+
+        return QStringLiteral("<img src=\"data:image/png;base64,%1\" width=%2 height=24 />")
+            .arg(QString::fromLatin1(buffer.data().toBase64())).arg(image.width());
```

Comment:
```suggestion
            .arg(QString::fromLatin1(buffer.data().toBase64()), image.width());
```

Reply:
This won't work - second argument is fillWidth:
![image](https://github.com/user-attachments/assets/81cf5ebd-1dff-42a5-98cc-5016e5e641c7)